### PR TITLE
docs(template-syntax/dart): updates to match TS

### PR DESCRIPTION
**Pull request by chalin**
Depends on #2750.

API references are still hard-coded to TS entries. This will be addressed in the scope of #1895.

- Propagates TS-side changes:
  - update #2639 - new two-way binding section, and
  - fix #2687 - invalid attr syntax
- Fixes
  - #1898 - currency symbols
  - #2748 - Dart template-syntax e2e is failing 
  - #2749 - deprecated `[className]`

### CODE REVIEW

**Comment by chalin** [comment]
Note to reviewers: you can ignore updates to TS files and `ts/latest` Jade files, since they will be reviewed via #2750.

cc @matanlurey @ferhatb

File: public/docs/ts/_cache/guide/template-syntax.jade [hunk 1]
```diff
@@ -742,47 +740,46 @@ block style-property-name-dart-diff
...
 .l-sub-section
   :marked
-    The `myClick` directive is further described below in the section
-    on [Aliasing input/output properties](#aliasing-io).
+    The `myClick` directive is further described in the section
+    on [aliasing input/output properties](#aliasing-io).
 
 :marked
-  If the name fails to match an element event or an output property of a known directive,
+  If the name fails to match element event or output property of a known directive,
```
Comment (kwalrath) [inline public/docs/ts/_cache/guide/template-syntax.jade]: I think the "an"s need to go back in.

File: public/docs/ts/_cache/guide/template-syntax.jade [hunk 2]
```diff
@@ -742,47 +740,46 @@ block style-property-name-dart-diff
...
-  To update the `firstName` property, we must get the changed text by following
-  the path `$event.target.value`.
+  To update the `firstName` property, the changed text is retrieved by following the path `$event.target.value`.
 
-  If the event belongs to a directive (remember: components are directives), `$event` has whatever shape the directive chose to produce.
+  If the event belongs to a directive (recall that components are directives), `$event` has whatever shape the directive decides to produce.
 
   <a id="eventemitter"></a>
   <a id="custom-event"></a>
   ### Custom Events with EventEmitter
```
Comment (kwalrath) [inline public/docs/ts/_cache/guide/template-syntax.jade]: Events -> events

GLOBAL: Headings should be sentence case. (There were lots of occurrences of title case that weren't visible in the diffs, so I haven't marked them.)

File: public/docs/ts/_cache/guide/template-syntax.jade [hunk 3]
```diff
@@ -843,19 +839,72 @@ block style-property-name-dart-diff
...
+:marked
+  ## Two-way binding
+  We often want to both display a data property and update that property when the user makes changes.
+
+  On the element side that takes a combination of setting a specific element property
+  and listening for an element change event.
+
+  Angular offers a special _two-way data binding_ syntax for this purpose, **`[(x)]`**.
+  The `[(x)]` syntax combines the brackets 
+  of _Property Binding_, `[x]`, with the parentheses of _Event Binding_, `(x)`.
```
Comment (kwalrath) [inline public/docs/ts/_cache/guide/template-syntax.jade]: Property Binding -> property binding
Event Binding -> event binding

(it looks like this section was added after the edit, and uses pre-edit capitalization patterns)

File: public/docs/ts/_cache/guide/template-syntax.jade (hunk 2 above)
Comment (chalin) [inline public/docs/ts/_cache/guide/template-syntax.jade]: Made a pass and fixed them all.

File: public/docs/ts/_cache/guide/template-syntax.jade (hunk 3 above)
Comment (chalin) [inline public/docs/ts/_cache/guide/template-syntax.jade]: fixed

**Comment by kwalrath** [comment]
LGTM. Let me know when it's safe to merge this.

**Comment by kwalrath** [comment]
Still LGTM. OK to merge once Travis passes?

**Comment by chalin** [comment]
#2750 should be merged in first. I've pinged @wardbell concerning that.

**Comment by kwalrath** [comment]
It looks like #2750 was merged, so once you resolve the conflicts, I'll merge this.

**Comment by chalin** [comment]
@kwalrath - PR is ready.

- Checked prose: no `BAD FILENAME`
- e2e pass:
```
Suites passed:
  public/docs/_examples/template-syntax/dart
  public/docs/_examples/template-syntax/ts
All tests passed
```

**Pull request merged**